### PR TITLE
Feature/analyze services

### PR DESCRIPTION
**Pull request by cdaringe**

### CODE REVIEW

File: app/main/services/analyze.js [hunk 1]
```diff
@@ -1,109 +1,46 @@
-var fs = require('fs');
+'use strict';
```
Comment (cdaringe) [inline app/main/services/analyze.js]: it looks like a lot changed here.  if you view the raw file, you'll see that really, it's very simple, but the IPC comm channel and the actual analysis bits were separated into different files

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "bootstrap": "twbs/bootstrap-sass#v3.3.5",
     "coinstac-distributed-algorithm-set": "git+https://github.com/MRN-Code/coinstac-distributed-algorithm-set.git",
     "config": "^1.14.0",
+    "errio": "^1.2.1",
```
Comment (cdaringe) [inline package.json]: You know how you cant `JSON.stringify` errors?  This ~enables you to do so.  i serialize errors from the main process during analysis for sending back to the render process for the UI to respond accordingly

File: webpack.config.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ module.exports = {
             ipc: 'commonjs ipc',
             config: 'commonjs config',
             sha: 'commonjs sha',
+            errio: 'commonjs errio',
             path: 'commonjs path',
             remote: 'commonjs remote'
```
Comment (cdaringe) [inline webpack.config.js]: as many things as we can we should stub out to externals to keep our bundle small.  what do you think?  all modules that don't need the window or its subcomponents

Comment (swashcap) [inline webpack.config.js]: There’s a Webpack configuration specifically for Electron. Probably useful to drop that in. We could also pipe all the `dependencies` keys from _package.json_ into here.

**Comment by swashcap** [comment]
This is okay for the most part. `FormManageProjectController`’s `handleAnalyze` should be split into action(s) and middleware to keep things Redux-y.

**Pull request merged**